src/js/BookReader.js

```diff
@@ -2978,207 +2974,63 @@ BookReader.prototype.jumpIndexForRightEdgePageX = function(pageX) {
-      self.trigger(BookReader.eventNames.stop);
-      self.$('.BRpageviewValue').val(window.location.href);
-    }
-  });
-  this.$('.info').colorbox({
-    inline: true,
-    opacity: "0.5",
-    href: this.$('.BRinfo'),
-    onLoad: function() {
-      self.trigger(BookReader.eventNames.stop);
-    }
-  });
-};
-
-BookReader.prototype.blankInfoDiv = function() {
-  return $([
-    '<div class="BRfloat BRinfo">',
-    '<div class="BRfloatHead">About this book',
-    '<button class="floatShut" href="javascript:;" onclick="$.fn.colorbox.close();"><span class="shift">Close</span></a>',
-    '</div>',
-    '<div class="BRfloatBody">',
-    '<div class="BRfloatCover">',
-    '</div>',
-    '<div class="BRfloatMeta">',
-    '<div class="BRfloatTitle">',
-    '<h2><a/></h2>',
-    '</div>',
-    '</div>',
-    '</div>',
-    '<div class="BRfloatFoot">',
-    '<a href="https://openlibrary.org/dev/docs/bookreader">About the BookReader</a>',
-    '</div>',
-    '</div>'].join('\n')
-  );
-};
-
-BookReader.prototype.blankShareDiv = function() {
-  return $([
-    '<div class="BRfloat BRshare">',
-    '<div class="BRfloatHead">',
-    'Share',
-    '<button class="floatShut" href="javascript:;" onclick="$.fn.colorbox.close();"><span class="shift">Close</span></a>',
-    '</div>',
-    '</div>'].join('\n')
-  );
-};
-
-/**
- * Update the displayed zoom factor based on reduction factor
- * @param {number} reduce
- */
-BookReader.prototype.updateToolbarZoom = function(reduce) {
-  var value;
-  var autofit = null;
-
-  // $$$ TODO preserve zoom/fit for each mode
-  if (this.mode == this.constMode2up) {
-    autofit = this.twoPage.autofit;
-  } else {
-    autofit = this.onePage.autofit;
-  }
 
-  if (autofit) {
-    value = autofit.slice(0,1).toUpperCase() + autofit.slice(1);
-  } else {
-    value = (100 / reduce).toFixed(2);
-    // Strip trailing zeroes and decimal if all zeroes
-    value = value.replace(/0+$/,'');
-    value = value.replace(/\.$/,'');
-    value += '%';
-  }
-  this.$('.BRzoom').text(value);
-};
+/************************/
+/** Toolbar extensions **/
+/************************/
+BookReader.prototype.buildToolbarElement = Toolbar.prototype.buildToolbarElement;
+exposeComponentMethod(Toolbar, 'toolbar', 'buildToolbarElement');
+BookReader.prototype.initToolbar = Toolbar.prototype.initToolbar;
+exposeComponentMethod(Toolbar, 'toolbar', 'initToolbar');
+BookReader.prototype.buildShareDiv = Toolbar.prototype.buildShareDiv;
+exposeComponentMethod(Toolbar, 'toolbar', 'buildShareDiv');
+BookReader.prototype.buildInfoDiv = Toolbar.prototype.buildInfoDiv;
+exposeComponentMethod(Toolbar, 'toolbar', 'buildInfoDiv');
+BookReader.prototype.getToolBarHeight = Toolbar.prototype.getToolBarHeight;
+exposeComponentMethod(Toolbar, 'toolbar', 'getToolBarHeight');
+/** @deprecated zoom no longer in toolbar */
+BookReader.prototype.updateToolbarZoom = Toolbar.prototype.updateToolbarZoom;
+exposeComponentMethod(Toolbar, 'toolbar', 'updateToolbarZoom');
```

Comment:
I was considering that, but I think that would break the autocompletes; right now we get autocompletion for these methods because of the `BookReader.prototype.Foo = Toolbar.prototype.Foo`; the type checker isn't smart enough to detect that through a for loop (I don't think).

Reply:
https://github.com/internetarchive/bookreader/issues/240